core/src/test/scala/org/zalando/grafter/RewriterSpec.scala

```diff
@@ -11,6 +11,7 @@ class RewriterSpec extends Specification with ThrownExpectations { def is = s2""
  An object graph can be rewritten with a node becoming a singleton object
    All identical nodes (having the same type) are replaced by just the first found instance $makeSingleton
    Singletons can be made according to a predicate                                          $makeSingletonPredicate
+   Making singletons must not fail on nested classes                                        $makeSingletonNestedClass
```

Comment:
Nested classes are for sure a problem but `org.http4s.Uri` is not. So I suspect the issue is with that class being final.